bob-apps/bin/bobp.rs

```diff
@@ -438,46 +529,70 @@ fn print_periodic_stat(
         DiffContainer::new(Box::new(|| stat.get_time_ns_st.load(Ordering::Relaxed)));
     let mut get_count_st =
         DiffContainer::new(Box::new(|| stat.get_count_st.load(Ordering::Relaxed)));
+    let mut exist_time_st =
+        DiffContainer::new(Box::new(|| stat.exist_time_ns_st.load(Ordering::Relaxed)));
+    let mut exist_count_st =
+        DiffContainer::new(Box::new(|| stat.exist_count_st.load(Ordering::Relaxed)));
     let mut get_size = DiffContainer::new(Box::new(|| stat.get_size_bytes.load(Ordering::Relaxed)));
+    let mut exist_size = DiffContainer::new(Box::new(|| stat.exist_size_bytes.load(Ordering::Relaxed)));
     let mut put_speed_values = vec![];
     let mut get_speed_values = vec![];
-    let k = request_bytes as f64 / period_ms as f64 * 1000.0 / 1024.0;
+    let mut exist_speed_values = vec![];
+    let sec = period_ms as f64 / 1000.;
+    let k = request_bytes as f64 / 1024.0;
     let start = Instant::now();
     while !stop_token.load(Ordering::Relaxed) {
         thread::sleep(pause);
-        let put_count_spd = put_count.get_diff() * 1000 / period_ms;
-        let get_count_spd = get_count.get_diff() * 1000 / period_ms;
-        let cur_st_put_time = put_time_st.get_diff() as f64;
-        let cur_st_put_count = put_count_st.get_diff() as f64;
-        let cur_st_get_time = get_time_st.get_diff() as f64;
-        let cur_st_get_count = get_count_st.get_diff() as f64;
-
-        let put_error = stat.put_error_count.load(Ordering::Relaxed);
-        let get_error = stat.get_error_count.load(Ordering::Relaxed);
-        let put_spd = put_count_spd as f64 * k;
-        let get_spd = get_size.get_diff() as f64 / 1024.0;
-        if put_spd > 0.0 {
+        let d_put = put_count.get_diff();
+        let d_get = get_count.get_diff();
+        let d_exist = exist_count.get_diff();
+        if d_put > 0 {
```

Comment:
Do I understand correctly that `d_put` is the number of `put` operations performed between two consecutive calls to `print_periodic_stat`? If yes, then this is incorrect condition, because if the performance of `put` operations is low, there can be 0 put requests executed in the time interval, but the statisitcs still should be printed.
It's better to check the `behaviour` parameter to see what to print